src/EditorFeatures/Core/Implementation/Classification/SyntacticClassificationTaggerProvider.TagComputer.cs

```diff
@@ -135,43 +149,72 @@ private void ConnectToWorkspace(Workspace workspace)
                 ResetLastParsedDocument();
 
                 _workspace = workspace;
-                _workspace.WorkspaceChanged += this.OnWorkspaceChanged;
-                _workspace.DocumentOpened += this.OnDocumentOpened;
-                _workspace.DocumentActiveContextChanged += this.OnDocumentActiveContextChanged;
 
-                var textContainer = _subjectBuffer.AsTextContainer();
+                if (_workspace != null)
+                {
+                    _isClassificationOnlyWorkspace = false;
+                    _workspace.WorkspaceChanged += this.OnWorkspaceChanged;
+                    _workspace.DocumentOpened += this.OnDocumentOpened;
+                    _workspace.DocumentActiveContextChanged += this.OnDocumentActiveContextChanged;
+
+                    var textContainer = _subjectBuffer.AsTextContainer();
 
-                var documentId = _workspace.GetDocumentIdInCurrentContext(textContainer);
-                if (documentId != null)
+                    var documentId = _workspace.GetDocumentIdInCurrentContext(textContainer);
+                    if (documentId != null)
+                    {
+                        var document = workspace.CurrentSolution.GetDocument(documentId);
+                        if (document != null)
+                        {
+                            EnqueueParseSnapshotTask(document);
+                            return;
+                        }
+                    }
+                }
+                else
                 {
-                    var document = workspace.CurrentSolution.GetDocument(documentId);
-                    if (document != null)
+                    // This buffer isn't being tracked by any workspace. In Visual Studio, this
+                    // can happen if it is not in the Running Document Table. We'll try to do
+                    // basic syntax classification anyway.
+
+                    _isClassificationOnlyWorkspace = true;
+                    var languageName = _contentTypesToLanguageNames.FirstOrDefault(x => _subjectBuffer.ContentType.MatchesAny(x.Metadata.DefaultContentType))?.Metadata.Language;
+                    if (languageName != null)
                     {
+                        _workspace = new AdhocWorkspace();
+                        var solution = _workspace.CreateSolution(SolutionId.CreateNewId());
+                        var project = solution.AddProject(name: string.Empty, assemblyName: string.Empty, language: languageName);
+                        var document = project.AddDocument(name: string.Empty, text: _subjectBuffer.CurrentSnapshot.AsText());
```

Comment:
Are we creating a workspace here only so we can call GetTreeAsync()? Should we just update EnqueueParseSnapshotTask to take a Lazy<Task<SyntaxTree>> or something else and so you can just call the parse yourself directly rather than having to create a workspace? Since you're not supporting this text changing (I'm still suspicious of that by the way), it might be easier.